pkg/datapath/loader/netlink.go

```diff
@@ -692,3 +694,33 @@ func renameDevice(from, to string) error {
 
 	return nil
 }
+
+// DeviceHasTCProgramLoaded checks whether a give device tc filter/qdisc progs are missing.
+func DeviceHasTCProgramLoaded(ifc string) (bool, error) {
+	l, err := netlink.LinkByName(ifc)
+	if err != nil {
+		return false, fmt.Errorf("unable to find endpoint link by name: %w", err)
+	}
+
+	dd, err := netlink.QdiscList(l)
+	if err != nil {
+		return false, fmt.Errorf("unable to fetch qdisc list for endpoint: %w", err)
+	}
+	var found bool
+	for _, d := range dd {
+		if d.Type() == qdiscClsact {
+			found = true
+			break
+		}
+	}
+	if !found {
+		return false, nil
+	}
+
+	ff, err := netlink.FilterList(l, netlink.HANDLE_MIN_INGRESS)
```

Comment:
Thank you for the review Robin! Initially I was checking both but after chatting with @christarazi he felt initially we only need to check the ingress side of the device with an option to open up egress since egress are managed by the daemon separately.

Reply:
To clarify, if it's possible to check for both ingress and egress then we should do it as Robin suggests. I recall that when Fernand and I were trying out `tc` commands to see what the TC filters looked like, we only found ingress. We assumed that egress was hidden in the endpoint's netns and therefore wasn't visible to Cilium in the host netns. However, we could totally be off base here, so @rgo3 it would be great if you can chime in and clarify.

Reply:
@rgo3 Added checks for egress when present.

Reply:
Great point! Thank you Robin. I'll update per your suggestions